app/models/decision.rb

```diff
@@ -1,50 +1,23 @@
 class Decision
   include ActiveModel::Model
 
-  attr_accessor :variant, :split, :admin
+  attr_accessor :split, :variant
 
-  validates :variant, presence: true, allow_blank: false
-  validates :split, presence: true
-  validates :admin, presence: true
-
-  validate :variant_belongs_to_split
+  validates :split, :variant, presence: true
 
   def save!
-    raise errors.full_messages.to_sentence unless valid?
-
-    weight_split_fully_to_variant
-    reassign_existing_assignments
-
-    true
-  end
-
-  def count
-    raise "count unavailable for unsaved Decision" unless bulk_assignment.persisted?
-    bulk_assignment.assignments.count
-  end
-
-  private
-
-  def weight_split_fully_to_variant
-    split.reweight!(variant => 100)
-  end
-
-  def reassign_existing_assignments
-    bulk_assignment.save!
-    existing_assignments.find_in_batches do |subset|
-      BulkReassignment.create!(assignments: subset, bulk_assignment: bulk_assignment)
+    raise "Variant must be present in the split" unless split.has_variant?(variant)
```

Comment:
Assuming the choice to inline the `variant_belongs_to_split` validation is because it's the only meaningful validation now?

Reply:
It was mostly the fact that I churned this file out of existence and back again. I'll put it back. It'd make it possible to use regular save in the controller too.

Reply:
Oh, actually it was to simplify the error bubbling - because I didn't need non-raisey save of decision, I figured I could skip all that and just make it blow up. Decision creations are never done by unwashed peeps - it'll either be admins or migration clients who should know better. That said, I may need real validations later to make the migration client spit actionable error messages. So I might be back to square one before I know it.

Reply:
Yep, totally fair. BTW if you were gonna go back to `raise errors.full_messages.to_sentence unless valid?` I would just change it to `validate!`.